src/hardware/BLEMIDI_ESP32_NimBLE.h

```diff
@@ -128,17 +128,17 @@ bool BLEMIDI_ESP32_NimBLE::begin(const char* deviceName, BLEMIDI_Transport<class
                                                      );
 
     _characteristic->setCallbacks(new MyCharacteristicCallbacks(this));
+
+    auto _security = new NimBLESecurity();
+    _security->setAuthenticationMode(ESP_LE_AUTH_BOND);
+
     // Start the service
     service->start();
-    
-    auto advertisementData = BLEAdvertisementData();
-    advertisementData.setFlags(0x04);
-    advertisementData.setCompleteServices(BLEUUID(SERVICE_UUID));
-    advertisementData.setName(deviceName);
 
     // Start advertising
     _advertising = _server->getAdvertising();
-    _advertising->setAdvertisementData(advertisementData);
+    _advertising->addServiceUUID(service->getUUID());
+    _advertising->setAppearance(0x00);
```

Comment:
this works nicely as well (apart from not setting the deviceName).
What makes this better than the previous version? (I must admit not to be a BLE expert)

Thanks @wakwak-koba